Sem. 03/Pract.03/Task.12.cpp

```diff
@@ -0,0 +1,35 @@
+
+#include <iostream>
+using namespace std;
+
+
+int main()
+{
+	int N;
+	cin >> N;
+	int num;
+	int sum = 0;
+	while (true) {
+		cin >> num;
+		if (num == - 1) {
+			break;
+		}
+		int copyOfnum = num;
+		int copyOfN = N;

```

Comment:
Here, before the algorithm for finding the gcd of the two numbers, you should check if (N < num) and swap them if so. Otherwise, potentially a lot of unnecessary calculations would happen.